core/src/main/java/org/bitcoinj/crypto/DeterministicEntropy.java

```diff
@@ -0,0 +1,221 @@
+     * @param language         Language <a href="https://bips.xyz/85#bip39">https://bips.xyz/85#bip39</a>.
+     * @param wordCount        Number of words in generated seed.
+     * @param index            Index of the child key.
+     * @return Seed for the BIP-85 derivation.
+     */
+    public static DeterministicSeed deriveBIP85Seed(DeterministicKey masterPrivateKey, Language language, WordCount wordCount, int index) {
+        // BIP-85 does not specify an upper limit on the index.  The index must be >= 0.
+        if (index < 0) throw new IllegalArgumentException("index must be >= 0");
+
+        // this is redundant because only a negative int would have the hardened bit set, but adding for clarity.
+        if (hasHardenedBit(index)) throw new IllegalArgumentException("index may not have hardened bit set");
+
+        HDPath.HDFullPath path = HDPath.m(BIP85_PATH_ROOT,
+                BIP39_APPLICATION_NUMBER,
+                language.childNumber(),
+                wordCount.childNumber(),
+                createHardenedChildNumber(index));
+
+        return deriveBIP85Seed(masterPrivateKey, path);
+    }
+
+
+    /**
+     * Perform a BIP-85 derivation and return the DeterministicSeed.
+     *
+     * @param masterPrivateKey DeterministicKey to derive from.
+     * @param wordCount        Number of words in generated seed.
+     * @param index            Index of the child key.
+     * @return Seed for the BIP-85 derivation.
+     */
+    public static DeterministicSeed deriveBIP85Seed(DeterministicKey masterPrivateKey, WordCount wordCount, int index) {
+        return deriveBIP85Seed(masterPrivateKey, Language.English, wordCount, index);
+    }
+
+    /**
+     * Derive the entropy for a BIP-85 derived key.
+     *
+     * @param masterPrivateKey DeterministicKey to derive from.
+     * @param hdPath           path of derivation.
+     * @return Entropy for the derived BIP-85 deterministic key.
+     */
+    private static byte[] deriveBIP85Entropy(DeterministicKey masterPrivateKey, HDPath hdPath) {
+        byte[] fullEntropy = deriveKey(masterPrivateKey, hdPath);
+        byte[] entropy512Bits = generateBIP85Entropy(fullEntropy);
+        byte[] entropy = Arrays.copyOf(entropy512Bits, WordCount.forWordCount(hdPath.children[3]).bitCount);
+
+        if (logger.isDebugEnabled()) {
+            HexFormat hex = new HexFormat();
+            logger.debug("deriveBIP85Entropy() BIP-85 Child Private Key (hex): {}", hex.formatHex(fullEntropy));
+            logger.debug("deriveBIP85Entropy() BIP-85 Entropy (hex)a: {}", hex.formatHex(entropy));
+            logger.debug("deriveBIP85Entropy() BIP-85 Entropy 512bits (hex): {}", hex.formatHex(entropy512Bits));
+            logger.debug("deriveBIP85Entropy() BIP-85 Entropy full (hex): {}", hex.formatHex(fullEntropy));
+        }
+
+        return entropy;
+    }
+
+    private static byte[] deriveKey(DeterministicKey childKey, HDPath hdPath) {
+        for (int val : hdPath.children) {
```

Comment:
Instead of .children[3] you should use `.list()` which returns a `List<ChildNumber>`.